src/engraving/layout/layoutmeasure.cpp

```diff
@@ -459,14 +459,28 @@ static bool breakMultiMeasureRest(const LayoutContext& ctx, Measure* m)
         return true;
     }
 
+    // Break for spanners/textLines in this measure
     auto sl = ctx.score()->spannerMap().findOverlapping(m->tick().ticks(), m->endTick().ticks());
     for (auto i : sl) {
         Spanner* s = i.value;
-        // break for first measure of volta or textline and first measure *after* volta
-        if ((s->isVolta() || s->isGradualTempoChange() || s->isTextLine()) && (s->tick() == m->tick() || s->tick2() == m->tick())) {
+        if ((s->isVolta() || s->isGradualTempoChange() || s->isTextLine())
+            && ((s->tick() >= m->tick() && s->tick() < m->endTick()) || (s->tick2() >= m->tick() && s->tick2() < m->endTick()))) {
             return true;
         }
     }
+    // Break for spanners/textLines starting or ending mid-way inside the previous measure
+    Measure* prevMeas = m->prevMeasure();
+    if (prevMeas) {
+        auto prevMeasSpanners = ctx.score()->spannerMap().findOverlapping(prevMeas->tick().ticks(), prevMeas->endTick().ticks());
+        for (auto i : prevMeasSpanners) {
+            Spanner* s = i.value;
+            if ((s->isVolta() || s->isGradualTempoChange() || s->isTextLine())
+                && ((s->tick2() > prevMeas->tick() && s->tick2() < prevMeas->endTick())
```

Comment:
Same as https://github.com/musescore/MuseScore/pull/15931/files#r1115733878

+ a lot of redundant calls (for example, prevMeas->tick() is called twice here). This also creates performance problems